accepted/00097-recurring-actions.md

```diff
@@ -0,0 +1,193 @@
+
+1) Create a `RecurringActionChainType` entity.
+2) Add an `action_chain_id` column to the `suseRecurringActionType` that would reference the `ActionChain` object.
+3) Add backend support for the ActionChainType.
+4) Adapt the `RecurringActionApplyJob` `Taskomatic` job to support the type.
+5) Allow CRUD operations to be performed on the frontend side.
+
+Recurring Action Chains could be accessed from the main menu `Schedule` -> ` Action Chains` -> `Recurring (Action Chains)`.
+
+### OpenSCAP scans
+
+To make openSCAP scans work properly with recurring actions we need to be able to create persistent openSCAP scan profiles that would be reused on every action execution. Currently openSCAP scans can be scheduled using a set of parameters and rescheduled only from the results of a previously performed scan.
+
+These profiles should also be able to apply to entire system groups and orgs to make recurring schedules on group/org level work.
+
+The `suseRecurringActionType` table would then only need to include an id referencing the openSCAP profile entity.
+
+The rest of the implementation would work similar to other action types.
+
+### Ansible Playbook execution
+
+TODO
+
+## Traditional Clients
+
+Even though traditional clients technically support openSCAP scans as well as action chains. I propose not to implement any recurring functionality for them. We want to stop supporting them and incentivize users to migrate to salt minions.
+
+## Maintenance Windows
+
+For recurring action scheduling to work properly the schedules needs to be aligned with existing maintenance windows on the clients. Otherwise the action is not going to be scheduled with the exception of actions allowed when the system is not in maintenance mode (openSCAP scans).
+
+## XMLRPC / HTTP API
+
+We already have a `RecurringActionHandler` that can be expanded to support other action types.
+
+I see two possibilities two implement new functions:
+
+1) For each type implement individiual CRUD function that states the action type in its name. E.g. `RecurringActionHandler.createCustomState(...)`. This is essentially just a wrapper that calls the generic `create` function with the action type as a parameter.
+
+2) Have one CRUD function for all the types and pass the action type as an argument.
+
+Option 1) could be more intuitive to use and its function would be clear just by reading its name, but add a lot of extra API endpoints that all call the same underlying base function, so option 2) would be less overhead.
+
+In the end I don't have a strong opinion which approach to take.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## SSM
+
+There are two ways recurring actions could work with SSM. However both come with significant downsides.
+
+1) Creating a recurrent action via SSM will schedule a `MinonRecurringAction` for every selected minion. This would schedule one `Taskomatic` job for every minion that would be scheduled to run at the same time.
```

Comment:
The primary advantage of SUSE Manager is its ability to allow users to perform tasks on multiple systems simultaneously. It is rare for users to perform actions on each system individually one by one.

But your concern is valid. If we can achieve the same result with a system group, then we can omit offering it as part of SSM.

Reply:
Remember: SSM is a temporary "group" and specific to the loged-in user.
What would work is:
SSM => create a group from current SSM selection => define a recuring action on that group. Right?
Recurring requires a permanent collection of systems. SSM is not permanent.

Reply:
Okay, I think we have an agreement here. Customers can just create a system group from their SSM selection and use it to create the recurring action. No need to go through the trouble to make it work with SSM